sdk/servicebus/azure-servicebus/samples/async_samples/deadletter_queue_async.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""
+Example to show receiving dead-lettered messages from a Service Bus Queue asynchronously.
+"""
+
+# pylint: disable=C0111
+
+import os
+import asyncio
+from azure.servicebus import ServiceBusMessage, ServiceBusSubQueue
+from azure.servicebus.aio import ServiceBusClient
+
+
+CONNECTION_STR = os.environ['SERVICE_BUS_CONNECTION_STR']
+QUEUE_NAME = os.environ["SERVICE_BUS_QUEUE_NAME"]
+
+async def send_messages(servicebus_client, num_messages):
+    sender = servicebus_client.get_queue_sender(queue_name=QUEUE_NAME)
+    async with sender:
+        msg = [
+            ServiceBusMessage(
+                "Message to be deadlettered: {}".format(i),
+                subject="{}".format("Bad" if i % 2 == 0 else "Good"),
+            )
+            for i in range(num_messages)
+        ]
+        await sender.send_messages(msg)
+        print("Messages sent")
+
+async def exceed_max_delivery(servicebus_client):
+    receiver = servicebus_client.get_queue_receiver(queue_name=QUEUE_NAME)
+    dlq_receiver = servicebus_client.get_queue_receiver(queue_name=QUEUE_NAME, 
+                                                        sub_queue=ServiceBusSubQueue.DEAD_LETTER,
+                                                        prefetch_count=10)
+    async with receiver:
+        peek_msgs = await receiver.peek_messages()
+        while len(peek_msgs) > 0:
+            received_msgs = await receiver.receive_messages()
+            for msg in received_msgs:
+                print("Picked up message: delivery_count {}".format(msg.delivery_count))
+                await receiver.abandon_message(msg)
+            peek_msgs = await receiver.peek_messages()
```

Comment:
interesting.. so we're using peek message as a condition to drain message in the queue.
this is migrated from the .net sample?

Reply:
The .net sample has a [`while True: receive if message not null; else break`](https://github.com/Azure/azure-service-bus/blob/c595bf8de42d41208a76adbe7329a8366466ed86/samples/DotNet/Microsoft.ServiceBus.Messaging/DeadletterQueue/Program.cs#L102) pattern, which isn't our pattern.
I used `len(peek_msgs)` here to check if messages were in the queue, rather than `len(received_msgs)` because `await receiver.receive_messages` was waiting forever. But I just realized that was because I didn't set `max_wait_time`. 🤦